src/util/getRandomValues.js

```diff
@@ -0,0 +1,55 @@
+// @flow
+
+/*
+ * API shim for `window.crypto.getRandomValues` in the browser, for
+ * compatibility with browser, Node, Jest, and Observable.
+ *
+ * Forked from NPM `get-random-bytes` by Kenan Yildirim, which is
+ * released under the MIT License.
+ */
+
+/**
+ * Fill the given buffer with cryptographically secure random bytes. The
+ * buffer length must not exceed 65536.
+ */
+function getRandomValues(buf: Uint8Array): Uint8Array {
+  if (typeof window !== "undefined") {
+    if (window.crypto && window.crypto.getRandomValues) {
+      return window.crypto.getRandomValues(buf);
+    }
+    if (
+      typeof window.msCrypto === "object" &&
+      typeof window.msCrypto.getRandomValues === "function"
+    ) {
+      return window.msCrypto.getRandomValues(buf);
+    }
+  }
+
+  if (typeof require !== "undefined") {
+    // Late-import `crypto` to avoid `require` in Observable notebooks,
+    // and avoid using a literal `require(...)` to prevent Webpack from
+    // rewriting this. (Getting Webpack externals to work properly is .)
+    //
+    /* eslint-disable camelcase */
+    /* eslint-disable no-undef */
+    const realRequire =
+      // $FlowExpectedError[cannot-resolve-name]
+      typeof __non_webpack_require__ !== "undefined"
+        ? __non_webpack_require__
+        : require; /* needed for Jest */
+    const nodeCrypto = realRequire("crypto");
+    if (buf.length > 65536) {
+      const e = new Error();
+      (e: any).code = 22;
```

Comment:
what is the significance of this? the `code` property isn't documented anywhere that I've seen

Reply:
Yoinked from `!npm get-random-values`, but the purpose is to emulate the
browser error. Node’s `crypto.randomBytes` lets you ask for any number
of random bytes, but in the browser it throws an error with this code
and name if you ask for more than 65536.